specutils/io/default_loaders/sdss_v.py

```diff
@@ -0,0 +1,573 @@
+                and len(hdulist) > 1 and (isinstance(hdulist[1], BinTableHDU))
+                and (hdulist[1].header.get("TTYPE3").lower() == "ivar"))
+
+
+def mwm_identify(origin, *args, **kwargs):
+    """
+    Check whether given input is FITS and has SDSS-V mwm type
+    BINTABLE in all 4 extensions. This is used for Astropy I/O Registry.
+    """
+    with read_fileobj_or_hdulist(*args, **kwargs) as hdulist:
+        return (("V_ASTRA" in hdulist[0].header.keys()) and len(hdulist) > 0
+                and ("SDSS_ID" in hdulist[0].header.keys())
+                and (isinstance(hdulist[i], BinTableHDU) for i in range(1, 5)))
+
+
+## HELPERS
+def _wcs_log_linear(naxis, cdelt, crval):
+    """
+    Convert WCS from log to linear.
+    """
+    return 10**(np.arange(naxis) * cdelt + crval)
+
+
+def _fetch_flux_unit(hdu):
+    """
+    Fetch the flux unit by accessing the BUNIT key of the given HDU.
+
+    Parameters
+    ----------
+    hdu : HDUImage
+        Flux or e_flux HDUImage object.
+
+    Returns
+    -------
+    flux_unit : Unit
+        Flux unit as Astropy Unit object.
+
+    """
+    flux_unit = (hdu.header.get("BUNIT").replace("(", "").replace(
+        ")", "").split(" ")[-2:])
+    flux_unit = "".join(flux_unit)
+    if "Ang" in flux_unit and "strom" not in flux_unit:
+        flux_unit = flux_unit.replace("Ang", "Angstrom")
+
+    flux_unit = flux_unit.replace("s/cm^2/Angstrom", "(s cm2 Angstrom)")
+
+    return Unit(flux_unit)
+
+
+## APOGEE files
+@data_loader(
+    "SDSS-V apStar",
+    identifier=apStar_identify,
+    dtype=Spectrum1D,
+    priority=10,
+    extensions=["fits"],
+)
+def load_sdss_apStar_1D(file_obj, idx: int = 0, **kwargs):
+    """
+    Load an apStar file as a Spectrum1D.
+
+    Parameters
+    ----------
+    file_obj : str, file-like, or HDUList
+        FITS file name, file object, or HDUList.
+    idx : int
+        The specified visit to load. Defaults to coadd (index 0).
+
+    Returns
+    -------
+    Spectrum1D
+        The spectrum contained in the file
+
+    """
+    # intialize slicer
+
+    with read_fileobj_or_hdulist(file_obj, memmap=False, **kwargs) as hdulist:
+        # Obtain spectral axis from header (HDU[1])
+        wavelength = _wcs_log_linear(
+            hdulist[1].header.get("NAXIS1"),
+            hdulist[1].header.get("CDELT1"),
+            hdulist[1].header.get("CRVAL1"),
+        )
+        spectral_axis = Quantity(wavelength, unit=Angstrom)
+
+        # Obtain flux and e_flux from data (HDU[1] & HDU[2])
+        flux_unit = _fetch_flux_unit(hdulist[1])
+        flux = Quantity(hdulist[1].data[idx], unit=flux_unit)
+        e_flux = StdDevUncertainty(hdulist[2].data[idx])
+
+        # reduce flux array if 1D in 2D np array
+        # NOTE: bypasses jdaviz specviz NotImplementedError, but could be the expected output for specutils
+        if flux.shape[0] == 1:
+            flux = flux[0]
+            e_flux = e_flux[0]
+
```

Comment:
not sure if we want any jdaviz specific logic in here since it's a downstream package and it could change the use this apStar spectrum1d outside of that context.

Reply:
Would this be its expected behavior anyways? If there's only a single spectrum within the data file, is the `flux` expected to still be loaded as an array within an array (2D, so without this block), or just a 1D array (with this block). Just wondering what the convention for this is.

Reply:
I think `Spectrum1D` just takes what you pass in as `flux`, so `np.array([1,2,3,4])`, and `[np.array([1,2,3,4])]` retain their shapes of `(4,)` and `(1,4)` respectively.   My point is only that we shouldn't put logic in here that is specifically addressing something in Jdaviz, but if this is the behaviour we want for the `Spectrum1D` object, then that's certainly ok.

Reply:
I think reducing a 2D flux array where one of the axes is degenerate to a 1D is reasonable, I would expect a 1D flux to come out the reader for a 1D spectrum.